megabeast/singlepop_dust_model.py

```diff
@@ -0,0 +1,241 @@
+        for cdust in dmod.keys():
+            cprior = dmod[cdust]["prior"]
+            if cprior["name"] != "fixed":
+                if cprior["name"] == "flat":
+                    if mval < cprior["m_min"] or mval > cprior["m_max"]:
+                        return -np.inf
+                    if mval < cprior["s_min"] or mval > cprior["s_max"]:
+                        return -np.inf
+                else:  # prior not supported
+                    exit()
+                    # raise warning instead
+        return 0.0
+
+
+def lnprob(phi, megabeast_model, beast_dust_priors, lnp_data, lnp_grid_vals):
+    """
+    Compute the log(likelihood) for the ensemble parameters
+
+    Parameters
+    ----------
+    phi: floats
+       ensemble parameters
+
+    megabeast_model : class
+        MegaBEAST physical model including priors
+
+    beast_dust_priors: PriorWeightsDust object
+       contains the data and functions for the dust ensemble model
+
+    lnp_data: dictonary
+       contains arrays of the lnp values and indexs to the BEAST model grid
+
+    lnp_grid_vals: dictonary
+       contains arrays of the beast parameters and priors for the sparse
+       lnp saved model grid points
+
+    Returns
+    -------
+    log(likelihood): float
+    """
+    ln_prior = megabeast_model.lnprior(phi)
+    print(ln_prior)
+    exit()
+
+    if not np.isfinite(ln_prior):
+        return -np.inf
+    return ln_prior + megabeast_model.lnlike(
+        phi, megabeast_model, beast_dust_priors, lnp_data, lnp_grid_vals
+    )
+
+
+def fit_ensemble(beast_data, lnp_filename, beast_priormodel, megabeast_model):
+    """
+    Run the MegaBEAST on a single set of BEAST results.
+
+    Parameters
+    ----------
+    beast_data : dict
+        information about the BEAST runs including SED grid and noise model
+
+    lnp_filename : string
+        file with posteriors from BEAST fitting
+
+    beast_priormodel : dict
+        BEAST prior model information
+
+    megabeast_model : dict
+        MegaBEAST physical model including priors
+
+    Returns
+    -------
+    fit_results : array
+        set of best fit parameters
+    """
+    # get the saved sparse likelihoods
+    lnp_data = read_lnp_data(lnp_filename, shift_lnp=True)
+
+    # get the completeness and BEAST model parameters for the
+    #   same grid points as the sparse likelihoods
+    lnp_grid_vals = get_lnp_grid_vals(beast_data, lnp_data)
+
+    # compute the BEAST prior weights
+    #  needed so the BEAST posteriors updated with the MegaBEAST model
+    # ***currently only AV ensemble model supported***
+    avs = lnp_grid_vals["Av"]
+    rvs = lnp_grid_vals["Rv"]
+    fAs = lnp_grid_vals["f_A"]
+    beast_dust_priors = PriorWeightsDust(
+        avs,
+        beast_priormodel["AV"],
+        rvs,
+        beast_priormodel["RV"],
+        fAs,
+        beast_priormodel["fA"],
+    )
+
+    # standard minimization to find initial values
+    def chi2(*args):
+        return -1.0 * lnprob(*args)
```

Comment:
Codacy found an issue: [No value for argument 'phi' in function call](https://app.codacy.com/manual/kgordon/megabeast/pullRequest?prid=6302069)